airflow/www/forms.py

```diff
@@ -82,7 +82,8 @@ class DagRunForm(DynamicForm):
         widget=Select2Widget())
     execution_date = DateTimeField(
         lazy_gettext('Execution Date'),
-        widget=AirflowDateTimePickerWidget())
+        widget=AirflowDateTimePickerWidget(),
+        format='%Y-%m-%d %H:%M:%S%z')
```

Comment:
It may be more work, but the ideal would be to use Pendulum to parse these dates -- that's what we use almost everywhere else in Airflow `airflow.utils.timezone.parse()`

Reply:
@ashb thing is `execution_date ` is a flask field. I am not sure how we can make flask use Pendulum.

Reply:
It _may_ be doable, but is not documented/clear. I think when I added timezones I got it working for display side (rather i got it using my custom control to decorate the fields so they get converted)

Reply:
@ashb created a custom field and use Pendulum now. Please check.